### PR TITLE
Bluetooth: controller: Tune tIFS switching

### PR DESCRIPTION
**Pull request by cvinayak**
Remove the 4us advanced radio reception, the implementation
passes all timing conformance tests without this.
This change should reduce some radio power consumption by
avoiding redundant reception duration.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c [hunk 1]
```diff
@@ -330,18 +329,20 @@ static void isr_tx(void *param)
...
 
+	radio_pkt_rx_set(radio_pkt_scratch_get());
 	/* assert if radio packet ptr is not set and radio started rx */
 	LL_ASSERT(!radio_is_ready());
 
 	if (IS_ENABLED(CONFIG_BT_CTLR_PROFILE_ISR)) {
 		lll_prof_cputime_capture();
 	}
 
+	radio_isr_set(isr_rx, param);
```
Comment (George-Stefan) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c]: @cvinayak Moving radio_isr_set() after radio_switch_complete_and_tx() makes any difference?

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan.c [hunk 2]
```diff
@@ -167,6 +166,7 @@ static int prepare_cb(struct lll_prepare_param *prepare_param)
 
 	radio_isr_set(isr_rx, lll);
```
Comment (George-Stefan) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan.c]: This is let before radio_switch_complete_and_tx()

**Comment by ioannisg** [comment]
I'd say, @cvinayak this needs to be tested on all nRF HW

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan.c [hunk 3]
```diff
@@ -466,6 +466,8 @@ static void isr_tx(void *param)
 				 radio_tx_chain_delay_get(0, 0) -
 				 CONFIG_BT_CTLR_GPIO_LNA_OFFSET);
 #endif /* CONFIG_BT_CTLR_GPIO_LNA_PIN */
+
+	radio_isr_set(isr_rx, param);
```
Comment (George-Stefan) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan.c]: This was moved after radio_switch_complete_and_tx()

**Comment by cvinayak** [comment]
> 
> 
> I'd say, @cvinayak this needs to be tested on all nRF HW

This is tested on nRF51, nRF52840 and nRF5340 cpunet, have run through conformance test suites for nRF51_pca10028 and nRF52840_pca10056 boards, no regression observed.

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c]: Couple CPU cycles, yes.

Comment (George-Stefan) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c]: @cvinayak  But is it related with the 4us you removed?

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c]: @George-Stefan 4us removed is in separate commit and refactoring to move non critical code later in the ISR is another commit.

**Review by George-Stefan — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**